app/screens/apps_form/apps_form_component.tsx

```diff
@@ -369,7 +364,11 @@ function AppsFormComponent({
                 return [];
             }
         }
-    }, []);
+    }, [form, values, performLookupCall, setErrors, intl]);
+
+    // for some reason pressing the button does not run the callback, so temporarily setting the close button to this callback
+    useNavButtonPressed(CLOSE_BUTTON_ID, componentId, handleSubmit, [handleSubmit]);
+    useNavButtonPressed(SUBMIT_BUTTON_ID, componentId, handleSubmit, [handleSubmit]);
```

Comment:
The `SUBMIT` button at the top right of the screen doesn't run the callback for some reason, so I've assigned the close button's callback to the submit handler for now